datafusion/physical-expr/src/aggregate/count_distinct/mod.rs

```diff
@@ -130,18 +132,22 @@ impl AggregateExpr for DistinctCount {
             Time64(Nanosecond) => {
                 Box::new(PrimitiveDistinctCountAccumulator::<Time64NanosecondType>::new())
             }
-            Timestamp(Microsecond, _) => Box::new(PrimitiveDistinctCountAccumulator::<
-                TimestampMicrosecondType,
-            >::new()),
-            Timestamp(Millisecond, _) => Box::new(PrimitiveDistinctCountAccumulator::<
-                TimestampMillisecondType,
-            >::new()),
-            Timestamp(Nanosecond, _) => Box::new(PrimitiveDistinctCountAccumulator::<
-                TimestampNanosecondType,
-            >::new()),
-            Timestamp(Second, _) => {
-                Box::new(PrimitiveDistinctCountAccumulator::<TimestampSecondType>::new())
-            }
+            dt @ Timestamp(Microsecond, _) => Box::new(
```

Comment:
I wonder if we should defensively always set `with_data_type` to `PrimitiveDistinctCountAccumulator` 🤔 

I tried to come up with other `DataType` that doesn't have a 1:1 mapping to its `ArrowPrimitiveType` and I don't think there is one, but always supplying the `data_type` might be a more defensive pattern

Reply:
I was wondering the same. Let me know what you prefer?

Reply:
How about we do it as a follow on PR? I would like to merge this one asap as it fixes a regression